### PR TITLE
Gettext support on Android

### PR DESCRIPTION
**Pull request by Pevernow**
Only based on https://github.com/minetest/minetest/pull/10019 ,I completed rebase to adapt to the new version.

Please refer to the original issue for all details.

Thank you for oleastre's work.

### CODE REVIEW

File: src/gui/modalMenu.cpp
Comment (sfan5) [inline src/gui/modalMenu.cpp:272]: ???

File: src/porting_android.cpp
Comment (sfan5) [inline src/porting_android.cpp:193]:
```suggestion
	path_locale  = path_share + DIR_DELIM + "locale";
```

File: src/gettext.cpp
Comment (sfan5) [inline src/gettext.cpp:215]: Can't `initializePathsAndroid` set this? That would be cleaner than adding more platform-specific code in here.

File: src/gui/modalMenu.cpp
Comment (Pevernow) [reply src/gui/modalMenu.cpp:272]: This is the code written by the original author, and I don't know why.

If there are no other problems, I will change this code back.

File: src/gettext.cpp
Comment (Pevernow) [reply src/gettext.cpp:215]: `initializePathsAndroid` is used to define the paths of minetest on Android platform.

What we do here is to get the system language and set it when the user does not have a custom language.

If `initializePathsAndroid` set this, the function will not match the function name, which will reduce the maintainability.

So I don't think `initializePathsAndroid` can do it.

Comment (sfan5) [reply src/gettext.cpp:215]: `initAndroid`, then

Comment (Pevernow) [reply src/gettext.cpp:215]: These two methods are reasonable, but in order to reduce the workload, I still do not modify this code

**Comment by Pevernow** [comment]
Fixed and enable gettext.

**Comment by Pevernow** [comment]
@sfan5  Do I need to add the function of automatically detecting the language environment?

File: src/gui/modalMenu.cpp
Comment (oleastre) [reply src/gui/modalMenu.cpp]: This was already asked in the original PR and answered there https://github.com/minetest/minetest/pull/10019#discussion_r437763396

`message` is of type `std::string`, and the type returned by `gettext(...)` is `char*`. 
And since we are adding `":"` just after, the compiler was considering that constant also as `char*`. But adding two `char*` is not supported. 
The conversion to `std::string` comes with the `+=` assignation
So this was the simplest way to just force the conversion of the two values to `std::string`.
Note that it was not needed before because `gettext` was just a macro returning the original type that was `std::string`.

File: src/gettext.cpp
Comment (oleastre) [reply src/gettext.cpp:215]: This seemed to be the correct place to do the `setenv` as all other platforms do the same here.
Can probably be done somewhere else, but in that case, it could be a re-factoring for all platforms no ?

File: src/gui/modalMenu.cpp
Comment (sfan5) [reply src/gui/modalMenu.cpp]: Huh so does the libintl on other platforms have C++ support?

File: src/gettext.cpp
Comment (sfan5) [reply src/gettext.cpp:215]: The issue is not setenv, the issue is that we have to pull in Android-specific code to retrieve the system language (which no other platform currently needs). It would be far easier to set the language unconditionally at startup to keep it contained within `porting.cpp`.

File: src/gui/modalMenu.cpp
Comment (oleastre) [reply src/gui/modalMenu.cpp]: I'm for it, as it seems more simple :)

Comment (Pevernow) [reply src/gui/modalMenu.cpp]: @sfan5  @oleastre Should I change the code with your style?

File: src/gettext.cpp
Comment (Pevernow) [reply src/gettext.cpp:215]: @oleastre 
Through my test, this code does not automatically detect the language environment.

I agree with @sfan5 that write a function in porting_android.cpp to return the language environment.

And call it in gettext.cpp.

File: src/gui/modalMenu.cpp
Comment (sfan5) [reply src/gui/modalMenu.cpp]: Yes

File: src/gettext.cpp
Comment (sfan5) [reply src/gettext.cpp:215]: So does the current state of this PR detect the language correctly?

Comment (Pevernow) [reply src/gettext.cpp:215]: It does not work.

**Comment by Pevernow** [comment]
@sfan5 

I've tested this PR and it works.

Currently, users can manually enable internationalization support by selecting a language.

I plan to assign the next PR to add the function of automatic detection system language.

But please merge this PR first. Because it's really available and it works.

File: src/gui/modalMenu.cpp
Comment (sfan5) [reply src/gui/modalMenu.cpp]: This still needs to be changed to `message += strgettext(label) + ":";`

**Comment by sfan5** [comment]
If this PR works otherwise please
* remove the non-functional language detection code
* ensure the CI works; it's missing the `gettext` package to correctly work ([script here](https://github.com/minetest/minetest/blob/master/.github/workflows/android.yml))

**Comment by Pevernow** [comment]
@sfan5  It has been revised according to your requirements.

**Comment by Pevernow** [comment]
@sfan5 It has been revised according to your requirements.
I've modified the CI code, but you haven't merged this https://github.com/minetest/minetest_android_deps_binaries/pull/3 , so CI doesn't work correctly.

**Review by sfan5 — APPROVED** [review]
LGTM

**Comment by Pevernow** [comment]
@rubenwardy @SmallJoker 
Merging this PR seems to require the approval of more core developers.

What do you think of this PR?

**Comment by Pevernow** [comment]
@sfan5 Can we merge this PR now?

**Comment by sfan5** [comment]
It needs two approvals.

**Comment by SmallJoker** [comment]
The changes look good, but the Android build in GH Actions currently fails. Will this resolve by itself on merge?

**Comment by Pevernow** [comment]
@SmallJoker 
image

As shown in the figure, it has been successful.

Please merge it, thank you

**Review by SmallJoker — APPROVED** [review]

**Comment by Pevernow** [comment]
It is ready to merge.

**Pull request merged**
